### DISCUSSION TITLE
uses-feature tag in android manifest file

### DISCUSSION THREAD

**Discussion by silencer07** [expo/expo#27267] [Proposals and Ideas]
Hi All!

My TV release was rejected because i needed to indicate that some hardware features are optional (i.e. uses-feature tag in android manifest)

How do I do this in managed flow of expo? https://developer.android.com/training/tv/start/hardware.html#unsupported-features0

thanks

**Comment by lukmccall** [comment]
Hi @silencer07, you can create a custom config plugin to add a custom `feature tag`. That plugin should modify the android manifest. [Here](https://docs.expo.dev/config-plugins/introduction/#modifying-the-androidmanifestxml) you can find more information about it.

**Reply by silencer07** [reply]
@lukmccall I came across that one, however some people do not have the necessary skillset to write one. I am basically trying to get a documented example on how to do this.

P.S. same question has been ask in SO but no one replied with concrete answer https://stackoverflow.com/questions/75013370/create-a-expo-config-plugin-file-to-modify-android-manifest

Maybe an example on how to do this with `withAndroidManifest` will do?

**Reply by lukmccall** [reply]
Here is a good example from the community - https://github.com/karlhorky/config-plugins/blob/9b22a71343d74d71353d8c6535e7d04da2fc4473/packages/react-native-ble-plx/src/withBLEAndroidManifest.ts#L56-L77.
That plugin is doing more work, but there's a part that adds the feature tag.

**Reply by julianD77** [reply]
I also found the React Native Google Cast plugin [here](https://github.com/expo/config-plugins/tree/main/packages/react-native-google-cast) useful to follow when needing to make changes to the Android Manifest

https://github.com/expo/config-plugins/blob/fa4397d502b20cf022e9564897343ef70f27896b/packages/react-native-google-cast/src/withAndroidGoogleCast.ts#L67
